index.js

```diff
@@ -24,6 +30,24 @@ app.use('/api', userRouter);
 app.use('/api', fieldOfActivityRouter);
 app.use('/api', companyRouter);
 app.use('/api', signinRouter);
+app.use('/api', (req, res, next) => {
+  const token = req.headers.authorization.split(' ')[1];
+  jwt.verify(token, privateKey, (err, decoded) => {
+    if (err) {
+      res.status(404).send({ message: 'User not found.' });
+    } else {
+      userRepository.get(decoded.id)
+        .then(user => {
+          if (user) {
+            req.user = user;
+            next();
+          } else {
+            res.status(404).send({ message: 'User not found.' });
```

Comment:
When the user is not found, you should return 401, because it means the user is not logged/authenticated.